components/brave_shields/browser/tracking_protection_service.cc

```diff
@@ -90,6 +104,83 @@ bool TrackingProtectionService::ShouldStartRequest(const GURL& url,
   return false;
 }
 
+void TrackingProtectionService::DispatchBlockedEvent(int render_process_id,
+  int render_frame_id, const GURL& request_url) {
+  DCHECK_CURRENTLY_ON(BrowserThread::UI);
+  RenderFrameHost* render_frame_host =
+      RenderFrameHost::FromID(render_process_id, render_frame_id);
+  int frame_tree_node_id = render_frame_host ?
+    render_frame_host->GetFrameTreeNodeId() : -1;
+  BraveShieldsWebContentsObserver::DispatchBlockedEvent(kTrackers, 
+    request_url.spec(), render_process_id, render_frame_id, frame_tree_node_id);
+}
+
+bool TrackingProtectionService::ShouldStoreState(HostContentSettingsMap* map, 
+  int render_process_id, int render_frame_id, const GURL& top_origin_url, 
+  const GURL& origin_url) {
+
+  if (!first_party_storage_trackers_initailized_) {
```

Comment:
this is a problem, we need to block until this is initialized and not just let things through. We should probably have a brave shields navigation throttle that blocks until all the shields data is loaded